.github/workflows/ci-actions.yml

```diff
@@ -21,6 +21,10 @@ jobs:
         with:
           version: v1.42
           args: --config=./.etc/golangci.yml ./...
+      - name: Check shadowing
+        run: |
+         go install golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow@latest
+         go vet -vettool=$(which shadow) ./... 2>&1 | awk -f .etc/action.awk
```

Comment:
golangci-lint is already supposed to check this, 
https://github.com/golangci/golangci-lint/blob/v1.42.0/pkg/golinters/govet.go#L158-L161

```
$ GOBIN=/tmp/bin go install -v golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow@v0.1.5
$ go vet -vettool=/tmp/bin/shadow ./dh/sidh/...
# github.com/cloudflare/circl/dh/sidh/internal/p503
dh/sidh/internal/p503/fp2_test.go:176:6: declaration of "quickCheckConfig" shadows declaration at line 71
dh/sidh/internal/p503/fp2_test.go:196:6: declaration of "quickCheckConfig" shadows declaration at line 71
# github.com/cloudflare/circl/dh/sidh/internal/p751
dh/sidh/internal/p751/fp2_test.go:176:6: declaration of "quickCheckConfig" shadows declaration at line 72
dh/sidh/internal/p751/fp2_test.go:196:6: declaration of "quickCheckConfig" shadows declaration at line 72
# github.com/cloudflare/circl/dh/sidh/internal/p434
dh/sidh/internal/p434/fp2_test.go:176:6: declaration of "quickCheckConfig" shadows declaration at line 67
dh/sidh/internal/p434/fp2_test.go:196:6: declaration of "quickCheckConfig" shadows declaration at line 67
```
(v0.1.6 is the latest, but the latest released version of golangci-lint still ships with v0.1.5)

weirdly enough I cannot get trigger the error:
```
$ docker run --rm -it -v $PWD:/app:ro -w /app -v $PWD/.etc/golangci.yml:/cfg.yml:ro golangci/golangci-lint:v1.42.1 golangci-lint run -v --config=/cfg.yml ./dh/sidh/...
INFO [config_reader] Used config file ../cfg.yml
INFO [lintersdb] Active 25 linters: [bodyclose deadcode depguard dogsled errcheck funlen goconst gocyclo gofmt goimports gosec gosimple govet ineffassign misspell nakedret staticcheck structcheck stylecheck typecheck unconvert unparam unused varcheck whitespace]
INFO [loader] Go packages loading at mode 575 (types_sizes|compiled_files|deps|exports_file|imports|files|name) took 1.7904349s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 21.9491ms
INFO [linters context/goanalysis] analyzers took 8.7494783s with top 10 stages: buildir: 2.035007s, goimports: 1.1625096s, gofmt: 572.8674ms, misspell: 476.0301ms, unparam: 353.604ms, buildssa: 349.8852ms, isgenerated: 252.9011ms, unconvert: 198.352ms, S1038: 191.114ms, gosec: 178.241ms
INFO [runner] Issues before processing: 83, after processing: 0
INFO [runner] Processors filtering stat (out/in): cgo: 83/83, skip_files: 83/83, exclude: 25/25, autogenerated_exclude: 25/83, exclude-rules: 0/25, filename_unadjuster: 83/83, path_prettifier: 83/83, skip_dirs: 83/83, identifier_marker: 25/25
INFO [runner] processing took 21.3313ms with stages: autogenerated_exclude: 18.8558ms, exclude-rules: 774.9µs, path_prettifier: 712.3µs, identifier_marker: 603.3µs, skip_dirs: 174.3µs, filename_unadjuster: 22.8µs, cgo: 16.6µs, max_from_linter: 13.7µs, skip_files: 13.7µs, source_code: 13.6µs, exclude: 13.6µs, uniq_by_line: 13.5µs, path_prefixer: 13.4µs, severity-rules: 13.1µs, path_shortener: 13µs, max_per_file_from_linter: 12.9µs, max_same_issues: 12.8µs, sort_results: 12.8µs, nolint: 12.8µs, diff: 12.4µs
INFO [runner] linters took 2.5318588s with stages: goanalysis_metalinter: 2.508974s
INFO File cache stats: 41 entries of total size 229.0KiB
INFO Memory: 45 samples, avg is 117.1MB, max is 258.9MB
INFO Execution took 4.355754s
```

Reply:
I hunted down and arrived to this variable. `vetxOnly`
https://cs.opensource.google/go/go/+/master:src/cmd/go/internal/work/action.go;drc=c7f2f51fed15b410dea5f608420858b401887d0a;l=502